### PR TITLE
[MI-3807]: Snackbar component

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**
## Ticket ID
[MI-3805](https://brightscout.atlassian.net/browse/MI-3805)

## Description
- Snackbar component

## Screenshots
<p>

</p>

### CODE REVIEW

File: webapp/src/components/Snackbar/Snackbar.component.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React, {useEffect, useRef} from 'react';
+import {useDispatch} from 'react-redux';
+
+import {Button} from '@brightscout/mattermost-ui-library';
+
+import {Icon} from 'components/Icon';
+
+import './Snackbar.styles.scss';
```
Comment (ayusht2810) [inline webapp/src/components/Snackbar/Snackbar.component.tsx:8]: fix order

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/components/Snackbar/Snackbar.component.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    const snackbarColorMap: Record<SnackbarColor, string> = {
+        error: 'var(--error-text)',
+        default: 'var(--center-channel-color)',
+        success: 'var(--online-indicator)',
+    };
+
+    return (
+        <div
+            className={'absolute bottom-20 right-20 left-20 py-8 px-12 rounded-4 d-flex gap-8 items-center justify-between elevation-2 msteams-sync-rhs__snackbar'}
```
Comment (muazzamqazi) [inline webapp/src/components/Snackbar/Snackbar.component.tsx:45]: Is there a container with `position: relative` which will always consistently nest this? Would fixed position be better fit for this scenario?

File: webapp/src/components/Snackbar/Snackbar.component.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return () => {
+            clearTimeout(timeId.current);
+        };
+    }, [isOpen]);
+
+    const snackbarColorMap: Record<SnackbarColor, string> = {
+        error: 'var(--error-text)',
+        default: 'var(--center-channel-color)',
+        success: 'var(--online-indicator)',
+    };
```
Comment (muazzamqazi) [inline webapp/src/components/Snackbar/Snackbar.component.tsx:41]: Are these colors not added to tw theme? 

If not, can we control this with classes (perhaps with modifier on `msteams-sync-rhs__snackbar`)? Or in any other way remove the inline styles

**Review by muazzamqazi — CHANGES_REQUESTED** [review]

File: webapp/src/components/Snackbar/Snackbar.component.tsx (hunk 3 above)
Comment (SaurabhSharma-884) [reply webapp/src/components/Snackbar/Snackbar.component.tsx:41]: These variables are defined in mattermost webapp

Comment (muazzamqazi) [reply webapp/src/components/Snackbar/Snackbar.component.tsx:41]: You can add classes that use and add the class modifiers conditionally based on props

File: webapp/src/containers/Rhs/Rhs.container.tsx [hunk 4]
```diff
@@ -33,8 +37,13 @@ export const Rhs = () => {
...
 
     return (
-        presentInWhitelist ?
-            <LinkedChannels/> :
-            <>{'MS Teams Sync plugin'}</>
+        <>
+            {
+                presentInWhitelist ?
+                    getRhsView() :
+                    <>{'MS Teams Sync plugin'}</>
```
Comment (muazzamqazi) [inline webapp/src/containers/Rhs/Rhs.container.tsx:44]:
```suggestion
                    'MS Teams Sync plugin'
```

File: webapp/src/reducers/snackbar/index.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export const snackbarSlice = createSlice({
+    name: 'snackbarState',
+    initialState,
+    reducers: {
+        showAlert: (state, {payload}: PayloadAction<SnackbarActionPayload>) => {
+            state.message = payload.message;
+            state.severity = payload.severity;
+            state.isOpen = true;
```
Comment (muazzamqazi) [inline webapp/src/reducers/snackbar/index.ts:18]:
```suggestion
            state = {...payload, isOpen: true};
```

**Review by muazzamqazi — CHANGES_REQUESTED** [review]

File: webapp/src/containers/Rhs/Rhs.container.tsx (hunk 4 above)
Comment (SaurabhSharma-884) [reply webapp/src/containers/Rhs/Rhs.container.tsx:44]: Done already in some PR

**Pull request merged**